thunder/executors/torch_autograd.py

```diff
@@ -94,6 +95,9 @@ def backward(ctx, *args):
         ctx.maybe_clear_saved_tensors()  # Delete the reference to all saved tensors in the context
         grads = ctx.compiled_backward([saved_tensors_list, ctx.saved_other], args)
 
+        if isinstance(ctx.compiled_backward, CUDAGraphExecutor):
+            saved_tensors_list.clear()
```

Comment:
Fine for now, but I am working on fixing this bit.

Reply:
BTW, why is `ctx.saved_other` is not treated similarly?

Reply:
It's not treated similarly because we care only about Tensor objects and CUDA memory usage and `ctx.saved_other` is most likely a tuple, not a list, to prevent accidental changes. We need to remove all possible references to Tensor objects as soon as possible during backward to not increase peak memory usage while computing the result of backward. `ctx.saved_other` includes all other objects that are not Tensors but still required to compute the backward function.
This memory optimization doesn't apply to CUDA Graphs because simple usage of CUDA Graphs expects static memory region to be available and used for all saved tensors from forward to backward, we can't free this piece of memory.
Clearing the list in this if statement doesn't have any effect on memory, it's only to skip the assert in the next line. I'd like to keep the assert because it ensures correct logic execution inside the generated backward Python code.

Reply:
Luckily, there is a solution that does not require doing so explicitly in this part of the code. I believe it should be agnostic to such things

Reply:
And we can still safe on things which are not static?

Reply:
Could you please create an issue for discussion summarizing what is problematic today and how you plan to address it?

Reply:
I will show in the PR. But the idea is very much what we discussed together. Having a storage that returns proxy objects with references to the storage with the ability to "clean"  memory once del is called upon them.